apps/checker/app/services/MatcherPool.scala

```diff
@@ -375,7 +375,7 @@ class MatcherPool(
     * available categories.
     */
   private def getApplicableCategories(query: Check) = query.categoryIds match {
-    case None      => getCurrentCategories.map(_.id)
-    case Some(ids) => ids
+    case Some(ids) if ids.size > 0 => ids
+    case _                         => getCurrentCategories.map(_.id)
```

Comment:
I wonder if leaving `categoryIds` as `undefined` if there aren't any suggestions (rather than assigning it to an [empty array](https://github.com/guardian/prosemirror-typerighter/blob/7fe8acffce01092c6f19ed7568a5fca7bc1d3c19/src/ts/components/Match.tsx#L50) in `prosemirror-typerighter`) would make the logic here more straightforward - we wouldn't have to check if the `ids.size` was bigger than 0. However it does feel more explicit to set `suggestions` to an empty array if there aren't any, rather than leaving it as `undefined`.

Reply:
We could definitely do that in `prosemirror-typerighter` - but I still think the expected behaviour in Typerighter should be to return all categories if the user requests `[]` - no-one will ever want to get no matches back. 

To me the existing pattern match felt misleading:
`Some(ids)` versus `None` felt like it should mean - they give me category ids and I use them, or they give me no ids, and I instead use all categories

Instead it was -
- They give me Some(ids), I use the ids given
- They give me Some(Set.empty), I use no categories (as they asked for a Set of no ids)
- They give me None - I include all categories

Reply:
Agreed! Ignore me, getting categories and suggestions confused - I blame the heat.

Reply:
Also agreed!